### PR TITLE
[BACK-2744] Add AnyHigh and AnyLow to Summaries

### PR DESCRIPTION
**Pull request by Roukoswarf**

### CODE REVIEW

File: data/summary/types/summary_cgm_test.go [hunk 1]
```diff
@@ -848,6 +857,15 @@ var _ = Describe("CGM Summary", func() {
 					Expect(userCGMSummary.Stats.Periods[periodKey].HasTimeInVeryHighPercent).To(BeTrue())
 					Expect(*userCGMSummary.Stats.Periods[periodKey].TimeInVeryHighPercent).To(Equal(0.200))
 
+					Expect(userCGMSummary.Stats.Periods[periodKey].HasTimeInVeryHighMinutes).To(BeTrue())
+					Expect(*userCGMSummary.Stats.Periods[periodKey].TimeInVeryHighMinutes).To(Equal(240 * v))
```
Comment (ewollesen) [inline data/summary/types/summary_cgm_test.go:861]: Consider using a different value here, as if you did typo between high and low, since you use the same expected value for both, your test might not catch it.

**Review by ewollesen — APPROVED** [review]
TL;DR Looks good. I had a couple of questions, but nothing show-stopping.

You mentioned in slack how typos in tags have been a problem. I didn't notice any in my review, but I wanted to point out that you can write tests to check for them. That is, you can write a bit of json/bson by hand, unmarshal it into a data type, and check that the values show up. Any typos in tag names would be detected. I don't see any such tests currently here.

File: data/summary/types/bgm.go [hunk 2]
```diff
@@ -337,11 +381,17 @@ func (s *BGMStats) CalculatePeriod(i int, offset bool, totalStats *BGMBucketData
 		HasTimeInVeryLowRecords: true,
 		TimeInVeryLowRecords:    pointer.FromAny(totalStats.VeryLowRecords),
 
+		HasTimeInAnyLowRecords: true,
```
[the remark this answers is not in the document]
Comment (Roukoswarf) [reply data/summary/types/bgm.go:384]: basically yes, theyre hardcoded for sorting nil always last, as mongo doesnt have nil high or nil low configuration.

File: data/summary/types/bgm.go [hunk 3]
```diff
@@ -62,6 +62,14 @@ type BGMPeriod struct {
 	TimeInVeryLowRecords      *int `json:"timeInVeryLowRecords" bson:"timeInVeryLowRecords"`
 	TimeInVeryLowRecordsDelta *int `json:"timeInVeryLowRecordsDelta" bson:"timeInVeryLowRecordsDelta"`
 
+	HasTimeInAnyLowPercent   bool     `json:"hasTimeInAnyLowPercent" bson:"hasTimeInAnyLowPercent"`
```
[the remark this answers is not in the document]
Comment (Roukoswarf) [reply data/summary/types/bgm.go:65]: We want these fields to default to false, not nil. The other records are distinctly different between 0 and nil, so they should be nullable.

[the remark this answers is not in the document]
Comment (ewollesen) [reply data/summary/types/bgm.go:65]: The "other places" I mentioned was in the clinic repo clinic/clinic/types.go
https://github.com/tidepool-org/platform/blob/anyhigh-anylow/vendor/github.com/tidepool-org/clinic/client/types.go#L971

[the remark this answers is not in the document]
Comment (toddkazakov) [reply data/summary/types/bgm.go:65]: ☝️ is auto generated which had influence on the underlying models - e.g. using pointers to booleans or slices :(

**Review by ewollesen — DISMISSED** [review]
Some small changes plus my logging additions (thanks Alex!)
